### PR TITLE
Improve error message

### PR DESCRIPTION
**Pull request by orangejenny**
http://manage.dimagi.com/default.asp?234102

Add module name to error message, since unique id is opaque to users.

@biyeun

### CODE REVIEW

File: corehq/apps/app_manager/suite_xml/sections/entries.py [hunk 1]
```diff
@@ -469,7 +469,7 @@ def get_target_module(case_type, module_id, with_product_details=False):
                     target = module.get_app().get_module_by_unique_id(module_id)
                     if target.case_type != case_type:
                         raise ParentModuleReferenceError(
-                            "Module with ID %s has incorrect case type" % module_id
+                            "Module '%s' with ID %s has incorrect case type" % (module.default_name(), module_id)
```
Comment (czue) [inline corehq/apps/app_manager/suite_xml/sections/entries.py]: good call. is it still useful to have the ID at all?

Comment (orangejenny) [inline corehq/apps/app_manager/suite_xml/sections/entries.py]: nah, I suppose not. updated.

**Comment by benrudolph** [comment]
nice, i recently ran into this when helping someone debug an app

**Comment by benrudolph** [comment]
heh, just looked at the ticket and that was the app i was trying to debug

**Pull request merged**